apps/meteor/app/lib/server/functions/notifications/messageContainsHighlight.ts

```diff
@@ -0,0 +1,21 @@
+import type { IMessage } from '@rocket.chat/core-typings';
+import { escapeRegExp } from '@rocket.chat/string-helpers';
+
+/**
+ * Checks if a message contains a user highlight
+ *
+ * @param {string} message
+ * @param {array|undefined} highlights
+ *
+ * @returns {boolean}
+ */
+export function messageContainsHighlight(message: Pick<IMessage, 'msg'>, highlights: string[] | undefined): boolean {
+	if (!highlights || highlights.length === 0) {
+		return false;
+	}
+
+	return highlights.some((highlight: string) => {
+		const regexp = new RegExp(`\\b${escapeRegExp(highlight)}\\b`, 'i');
```

Comment:
I'm thinking, this will still make us highlight emojis (`:highlight:`), won't it? I wonder if we should take the chance to fix this too (since they're not exactly words also)

Reply:
hmmm.. good point.. let me create a few unit tests

Reply:
updated to not match emojis.. not the prettiest regex but that's the one I could come up with